grantPermission.ps1

```diff
@@ -0,0 +1,274 @@
+                IsError = $true
+            })
+
+        # Throw terminal error
+        throw
+    }
+
+    # Grant permission
+    try {
+        Write-Verbose "Granting permission to $($pRef.Name) ($($pRef.id)) for $($currentAccount.userName) ($($currentAccount.id))"
+
+        # Create permission body
+        $permissionBody = [PSCustomObject]@{
+            schemas    = "urn:ietf:params:scim:schemas:core:2.0:Group"
+            id         = $pRef.id
+            operations = @(
+                @{
+                    op    = "add"
+                    path  = "members"
+                    value = @(
+                        @{
+                            value   = $currentAccount.id
+                            display = $currentAccount.userName
+                        }
+                    )
+                }
+            )
+        }
+
+        $body = ($permissionBody | ConvertTo-Json -Depth 10)
+        $splatWebRequest = @{
+            Uri             = "$baseUrl/scim/groups/$($pRef.Id)"
+            Headers         = $headers
+            Method          = 'PATCH'
+            Body            = ([System.Text.Encoding]::UTF8.GetBytes($body))
+            ContentType     = "application/json;charset=utf-8"
+            UseBasicParsing = $true
+        }
+
+        if (-Not($actionContext.DryRun -eq $true)) {
+            Write-Verbose "Granting permission: [$($pRef.Name) ($($pRef.id))] to account: [$($currentAccount.userName) ($($currentAccount.id))]"
+
+            $addPermission = Invoke-RestMethod @splatWebRequest -Verbose:$false
+
+            $outputContext.AuditLogs.Add([PSCustomObject]@{
+                    # Action  = "" # Optional
+                    Message = "Successfully granted permission: [$($pRef.Name) ($($pRef.id))] to account: [$($currentAccount.userName) ($($currentAccount.id))]"
+                    IsError = $false
+                })
+        }
+        else {
+            Write-Warning "DryRun: Would grant permission: [$($pRef.Name) ($($pRef.id))] to account: [$($currentAccount.userName) ($($currentAccount.id))]"
+        }
+    }
+    catch {
+        $ex = $PSItem
+        $errorMessage = Get-ErrorMessage -ErrorObject $ex
+
+        Write-Verbose "Error at Line [$($ex.InvocationInfo.ScriptLineNumber)]: $($ex.InvocationInfo.Line). Error: $($errorMessage.VerboseErrorMessage)"
+        Write-Verbose "URI: $($splatWebRequest.Uri)"
+        Write-Verbose "Body: $($body)"
+
+        $outputContext.AuditLogs.Add([PSCustomObject]@{
+                # Action  = "" # Optional
+                Message = "Error granting permission: [$($pRef.Name) ($($pRef.id))] to account: [$($currentAccount.userName) ($($currentAccount.id))]. Error Message: $($errorMessage.AuditErrorMessage)"
+                IsError = $true
+            })
+
+        # Throw terminal error
+        throw
+    }
+}
+catch {
+    $ex = $PSItem
```

Comment:
do we need to log this in the auditlogs?

Reply:
No, only needed for dryrun